File: application/src/main/java/org/thingsboard/server/service/export/DataExportService.java
package main.java.org.thingsboard.server.service.export;

import org.thingsboard.server.common.data.export.DataExportPreviewRequest;
import org.thingsboard.server.common.data.export.DataExportPreviewResponse;
import org.thingsboard.server.common.data.export.DataExportRequest;
import org.thingsboard.server.common.data.export.DataExportSchedule;
import org.thingsboard.server.common.data.export.DataExportScheduleRequest;
import org.thingsboard.server.service.security.model.SecurityUser;

import java.io.OutputStream;

public interface DataExportService {

    DataExportPreviewResponse preview(SecurityUser user, DataExportPreviewRequest request) throws Exception;

    void writeCsv(SecurityUser user, DataExportRequest request, OutputStream outputStream) throws Exception;

    void saveSchedule(SecurityUser user, DataExportScheduleRequest request) throws Exception;

    DataExportSchedule getSchedule(SecurityUser user) throws Exception;

    void runSchedules() throws Exception;
}

File: ui-ngx/src/app/modules/home/pages/data_export/data-export.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {
  DataExportPreviewRequest,
  DataExportPreviewResponse,
  DataExportRequest,
  DataExportScheduleRequest
} from './data-export.models';

@Injectable({
  providedIn: 'root'
})
export class DataExportService {

  constructor(private http: HttpClient) {}

  getInitialPreview(): Observable<DataExportPreviewResponse> {
    return this.http.post<DataExportPreviewResponse>(
      '/api/data-export/preview',
      {
        deviceIds: [],
        includeCalculatedFields: true,
        includeAttributes: true
      }
    );
  }

  preview(request: DataExportPreviewRequest): Observable<DataExportPreviewResponse> {
    return this.http.post<DataExportPreviewResponse>('/api/data-export/preview', request);
  }

  exportCsv(request: DataExportRequest): Observable<Blob> {
    return this.http.post('/api/data-export/csv', request, {
      responseType: 'blob' as 'json'
    }) as Observable<Blob>;
  }

  saveSchedule(request: DataExportScheduleRequest): Observable<any> {
    return this.http.post('/api/data-export/schedule', request);
  }

  getSchedule(): Observable<any> {
    return this.http.get('/api/data-export/schedule');
  }
}